### PR TITLE
refactor: Renames user roles, improves security

### LINKED ISSUE #242: Tweak roles: Add "Director" position, remove "Admin"
Old structure:
User -> `event_tracking` -> `admin_limited_access` -> Admin

New structure:
User -> Volunteer -> Organizer -> Director

Directors can modify the permissions of organizers, while organizers cannot modify other organizers' permissions.

This will allow for more role customization down the road.

### PR DESCRIPTION
**Pull request by cbaudouinjr**
- Renames admin->Director, admin_limited_access->Organizer, event_tracking->Volunteer
- Redesigns Volunteer manage area
- Prevents Volunteers from seeing Dashboard (hides hackathon application numbers and other sensitive statistics, also simplifies the UI for Voluneteers who are usually first-time users)

Volunteer sidebar:
image

- Refactors underlying Manage security (security on a per-feature basis) 
- Staff members trying to access a restricted area will be redirected to their default Manage area rather than being kicked out of Manage (Dashboard for Directors, Organizers and Check-in for Volunteers)

closes #242

### CODE REVIEW

**Comment by cbaudouinjr** [comment]
Known issues:

- [ ] For volunteer role, "Check-in" button on sidebar is not highlighted by default

**Comment by peterkos** [comment]
Add'l bug: duplicate item for "check in" on director role maybe?

File: docs/messages.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ Automated emails can also be sent out upon certain events happening. All events
 * **Questionnaire status** &mdash; Upon being accepted, denied, RSVP'd, etc
 * **Bust list** &mdash; Becoming a passenger or bus captain
 
-These messages are sent immediatley upon an applicant entering the given state. For example, when an admin marks someone as "accepted," they will immediately receive any automated messages associated with the "Questionnaire Status: Accepted" event.
+These messages are sent immediatley upon an applicant entering the given state. For example, when a staff member marks someone as "accepted," they will immediately receive any automated messages assocaited with the "Questionnaire Status: Accepted" event.
```
Comment (peterkos) [inline docs/messages.md:56]: We need to define "staff member" somewhere in the docs; I'll put this in 2.0 migration guide for now

**Review by peterkos — APPROVED** [review]

**Comment by peterkos** [comment]
Merged in https://github.com/codeRIT/hackathon-manager/commit/58b58492cc2a37555c05c222ef69670023e4716f due to GitHub glitch in merge conflict; will address fixes in new PR

**Pull request closed without merging**